Update Readme to include authentication details (#19)
TimefoldAI/timefold-solver-enterprise: README.adoc

== Using Timefold Solver Enterprise Edition

To use Timefold Solver Enterprise Edition, first https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens[create a GitHub access token] with the `read:packages` permission.

> NOTE: Please note that the next version of Timefold Solver Enterprise Edition will not require any authentication and thus, the entire configuration procedure is going to be more straightforward.

Reviewer: ```suggestion
NOTE: In the near future, we will introduce anonymous access to _Timefold Solver Enterprise Edition_ packages. This step will no longer be required.
```

> === Using Timefold Solver Enterprise Edition in Maven

Reviewer: ```suggestion
=== 1a: Using Timefold Solver Enterprise Edition in Maven
```

      </repositories>
      ...
    </project>
+
Note that the repository `ID` is the same as the server `ID` from the previous step.

> === Using Timefold Solver Enterprise Edition with Gradle

Reviewer: ```suggestion
=== 1b: Using Timefold Solver Enterprise Edition with Gradle
```

      </repositories>
      ...
    </project>
+
Note that the repository `ID` is the same as the server `ID` from the previous step.

=== Step 1b: Using Timefold Solver Enterprise Edition with Gradle

Configure your Gradle project to use the Enterprise Edition Maven repository:

    repositories {
      mavenCentral()
      maven {
        url "https://maven.pkg.github.com/TimefoldAI/timefold-solver-enterprise"
        credentials {
            username = "GitHub user name"
            password = "GitHub token"
        }
        authentication {
            basic(BasicAuthentication)
        }
      }
    }

> === Step 2: Replace Community Edition artifacts

Reviewer: ```suggestion

=== Step 2: Replace Community Edition artifacts
```
